### ISSUE TITLE
[Feature] Add Custom Image Numbering and Sorting for Galleries

### ISSUE THREAD

**Issue by samboston208** [stashapp/stash#4950]
In Stash, users can organize scenes in a custom order using the "scene number" feature in the movies tab. This offers great flexibility, because it's possible to sort by the scene number.

When it comes to organizing images, Stash provides the option to create galleries. However, unlike the scene number feature for movies, galleries lack a similar system for numbering images. I was trying to do the same thing using chapters, but they're quite unreliable. As images are added or removed, image numbers change, screwing all the chapters up. Also, there is no option to sort by chapters.

My proposition is introducing an "image number" feature for galleries, similar to the scene number in movies. This feature should include:

The ability to manually set image numbers.
An option to sort images by their assigned numbers.

**Comment by Valkyr-JS** [comment]
Agreed, would very much like this. Most of my gallery images have their index in the filename and are ordered correctly, but this isn't always the case. I'd like to not have to depend on the filename for ordering.

**Comment by GalacticCheat** [comment]
Maista sucks cock
![Image](https://github.com/user-attachments/assets/ed12369e-cb07-434a-b49f-fbc470272035)
https://i.ibb.co/ds49xcXt/image-2-copie.jpg

**Comment by jtjb07** [comment]
I'd also suggest linking chapters to something like the image's ID for two reasons:
- It'd solve the problem described in the OP of chapters pointing to the wrong image when new images are added to the gallery
- It'd allow chapters to be used with ordering other than filenames (e.g. image title), without having to manually set image numbers, for those of us who'd rather use the existing sorting options to order our galleries
